sql/layers/layers.go

```diff
@@ -95,6 +95,55 @@ func UnsetAppliedFrom(db sql.Executor, lid types.LayerID) error {
 	return nil
 }
 
+// UpdateStateHash for the layer.
+func UpdateStateHash(db sql.Executor, lid types.LayerID, hash types.Hash32) error {
+	if _, err := db.Exec(`insert into layers (id, state_hash) values (?1, ?2) 
```

Comment:
this is slightly confusing. this implies that we may update state hash before we set the applied_block?
i.e. if we insert a new row here, this is a programming error.
what do you think just do `update layers set state_hash where id = lid`?

Reply:
i guess it would make sense to update block and hash at the same time, but lets switch to update for now

Reply:
actually let me keep it like this in this change. i will shortly start working on https://github.com/spacemeshos/go-spacemesh/issues/1798 , and i will try to update state hash and block for the layer at the same time